lib/reporter.js

```diff
@@ -25,10 +25,43 @@ class AllureReporter extends events.EventEmitter {
 
         const { epilogue } = this.baseReporter
 
+        this.testFrameworkEvents()
+        this.hooksEvents()
+        this.runnerEvents()
+        this.allureEvents()
```

Comment:
I think it is a good idea to move this into functions however from the unit test perspective let's keep the event listener here and move everything that happens on these events into dedicated functions. So instead of:
```js
this.allureEvents()
```
Let's do:
```js
this.on('allure:feature', ::this.allureFeature)
// ...
allureFeature ({ cid, featureName }) {
    const allure = this.getAllure(cid)
    const test = allure.getCurrentTest()
    test.addLabel('feature', featureName)
}
```

Reply:
Sure, I will change it. I just want to group this function by caller